sdk/keyvault/Azure.Security.KeyVault.Shared/Argument.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+
+namespace Azure.Security.KeyVault
+{
+    internal static class Argument
```

Comment:
>Argument [](start = 26, length = 8)

Perhaps this should be called something like AssertArgument instead of just Argument? Is it strange that Argument.NotNull throws an ArgumentNullException?  Just a thought.